File: cadwiki-nuget/cadwiki.NUnitTestRunner/UI/WpfDriver.vb
Option Strict On
Option Infer Off
Option Explicit On

Imports cadwiki.NUnitTestRunner.Results
Namespace Ui
    Public Class WpfDriver

        Private _regressionTestTypes As Type()
        Private _window As WindowTestRunner = Nothing
        Private _commonUiObject As New CommonUiObject()

        Public Function GetWindow() As WindowTestRunner
            Return _window
        End Function

        Public Sub New()
            _window = New WindowTestRunner()
        End Sub

        Public Sub New(ByRef suiteResult As ObservableTestSuiteResults,
                       regressionTestTypes As Type())
            Try
                If suiteResult Is Nothing Then
                    suiteResult = New ObservableTestSuiteResults()
                End If
                If regressionTestTypes Is Nothing Then
                    Console.WriteLine("RegressionTestTypes argument was null.")
                    Return
                End If
                _regressionTestTypes = regressionTestTypes
                _window = New WindowTestRunner(suiteResult)
                _commonUiObject.WpfAddResultsToTreeView(suiteResult, _window.TreeViewResults)
            Catch ex As Exception
                Dim window As cadwiki.WpfUi.Templates.WindowAutoCADException =
                    New WpfUi.Templates.WindowAutoCADException(ex)
                window.ShowDialog()
            End Try
        End Sub

        Public Async Function ExecuteTestsAsync() As Task
            Try
                Dim stopWatch As Stopwatch = New Stopwatch()
                stopWatch.Start()
                Await Engine.RunTestsFromType(_window.ObservableResults, stopWatch, _regressionTestTypes)
                _commonUiObject.WpfAddResultsToTreeView(_window.ObservableResults, _window.TreeViewResults)
            Catch ex As Exception
                Dim window As cadwiki.WpfUi.Templates.WindowAutoCADException =
                    New WpfUi.Templates.WindowAutoCADException(ex)
                window.ShowDialog()
            End Try

        End Function
    End Class
End Namespace

File: cadwiki-nuget/cadwiki.NUnitTestRunner/UI/WpfDriver.cs
using System;
using System.Diagnostics;
using System.Threading.Tasks;
using cadwiki.NUnitTestRunner.Results;

namespace cadwiki.NUnitTestRunner.Ui
{
    public class WpfDriver
    {

        private Type[] _regressionTestTypes;
        private cadwiki.NUnitTestRunner.UI.WindowTestRunner _window = null;
        private CommonUiObject _commonUiObject = new CommonUiObject();

        public cadwiki.NUnitTestRunner.UI.WindowTestRunner GetWindow()
        {
            return _window;
        }

        public WpfDriver()
        {
            _window = new cadwiki.NUnitTestRunner.UI.WindowTestRunner();
        }

        public WpfDriver(ref ObservableTestSuiteResults suiteResult, Type[] regressionTestTypes)
        {
            try
            {
                if (suiteResult is null)
                {
                    suiteResult = new ObservableTestSuiteResults();
                }
                if (regressionTestTypes is null)
                {
                    Console.WriteLine("RegressionTestTypes argument was null.");
                    return;
                }
                _regressionTestTypes = regressionTestTypes;
                _window = new cadwiki.NUnitTestRunner.UI.WindowTestRunner(ref suiteResult);
                _commonUiObject.WpfAddResultsToTreeView(suiteResult, _window.TreeViewResults);
            }
            catch (Exception ex)
            {
                var window = new WpfUi.Templates.WindowAutoCADException(ex);
                window.Show();
            }
        }

        public async Task ExecuteTestsAsync()
        {
            try
            {
                var stopWatch = new Stopwatch();
                stopWatch.Start();
                await Engine.RunTestsFromType(_window.ObservableResults, stopWatch, _regressionTestTypes);
                _commonUiObject.WpfAddResultsToTreeView(_window.ObservableResults, _window.TreeViewResults);
            }
            catch (Exception ex)
            {
                var window = new WpfUi.Templates.WindowAutoCADException(ex);
                window.Show();
            }

        }
    }
}